DDDEastAnglia.Tests/Helpers/LoginMethods/LoginMethodIconProviderTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using DDDEastAnglia.Helpers.AppSettings;
+using DDDEastAnglia.Helpers.LoginMethods;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace DDDEastAnglia.Tests.Helpers.LoginMethods
+{
+    [TestFixture]
+    public sealed class LoginMethodIconProviderTests
+    {
+        [Test]
+        public void GetIcon_ReturnsTheConfiguredIconForOAuthProvider()
+        {
+            var appSettingsProvider = Substitute.For<IAppSettingsProvider>();
+            appSettingsProvider.GetSetting(LoginMethodIconProvider.AppSettingsKeyPrefix + "test").Returns("test-icon");
```

Comment:
Ok. Could we factor that logic out of the tests then, so they're clearer to read? Like an `AppSettingsProviderBuilder`, perhaps (or just a Factory Method). Actually, it seems to me as though the AppSettingsProvider could just return `"test-icon"` for all input and the test would still pass - that's fine (we're testing that the SUT uses the return value from the Substitute correctly), and embracing it allows us to remove cruft from the test.